src/NorthwindWebsite/Core/EmailSender/EmailSender.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Microsoft.AspNetCore.Identity.UI.Services;
+using Microsoft.Extensions.Options;
+using SendGrid;
+using SendGrid.Helpers.Mail;
+
+namespace NorthwindWebsite.Core.EmailSender;
+
+public class EmailSender : IEmailSender
+{
+    public EmailSender(IOptions<AuthMessageSenderOptions> optionsAccessor)
+    {
+        Options = optionsAccessor.Value;
+    }
+
+    public AuthMessageSenderOptions Options { get; } //set only via Secret Manager
+
+    public Task SendEmailAsync(string email, string subject, string message)
+    {
+        return Execute(Options.SendGridApiKey, subject, message, email);
+    }
+
+    public Task Execute(string apiKey, string subject, string message, string email)
+    {
+        var client = new SendGridClient(apiKey);
+        var msg = new SendGridMessage()
+        {
+            From = new EmailAddress("evg.tsarkov@gmail.com", "Northwind Training App"),
+            Subject = subject,
+            PlainTextContent = message,
+            HtmlContent = message
+        };
+        msg.AddTo(new EmailAddress(email));
+
+        // Disable click tracking.
+        // See https://sendgrid.com/docs/User_Guide/Settings/tracking.html
+        msg.SetClickTracking(false, false);
+
+        return client.SendEmailAsync(msg);
+    }
+}
+
+public class AuthMessageSenderOptions
```

Comment:
Can we make this class nested in `EmailSender` or move to separate file?
Try to make only single class inside of c# file